core/src/main/java/org/testcontainers/containers/output/FrameConsumerResultCallback.java

```diff
@@ -78,4 +104,79 @@ public void close() throws IOException {
     public CountDownLatch getCompletionLatch() {
         return completionLatch;
     }
+
+    private synchronized void processRawFrame(OutputFrame outputFrame, Consumer<OutputFrame> consumer) {
+        if (outputFrame != null) {
+            String utf8String = outputFrame.getUtf8String();
+            byte[] bytes = outputFrame.getBytes();
+
+            if (utf8String != null && !utf8String.isEmpty()) {
+                // Merging the strings by bytes to solve the problem breaking non-latin unicode symbols.
+                if (brokenFrame != null) {
+                    bytes = merge(brokenFrame.getBytes(), bytes);
+                    utf8String = new String(bytes);
+                    brokenFrame = null;
+                }
+                // Logger chunks can break the string in middle of multibyte unicode character.
+                // Backup the bytes to reconstruct proper char sequence with bytes from next frame.
+                if (Character.getType(utf8String.charAt(utf8String.length() - 1)) == Character.OTHER_SYMBOL) {
```

Comment:
I would like `Character.getType(utf8String.charAt(utf8String.length() - 1))` in a variable.